### PR TITLE
Test support for stubs compiled with Scala 3 & publish codegen for Scala 3

### PR DESCRIPTION
**Pull request by bjaglin**

### CODE REVIEW

File: project/plugins.sbt [hunk 1]
```diff
@@ -1,6 +1,6 @@
 addSbtPlugin("com.thesamet" % "sbt-protoc" % "1.0.4")
 
-libraryDependencies += "com.thesamet.scalapb" %% "compilerplugin" % "0.11.1"
+libraryDependencies += "com.thesamet.scalapb" %% "compilerplugin" % "0.11.4"
```
Comment (bjaglin) [inline project/plugins.sbt:3]: https://github.com/scalapb/ScalaPB/issues/1174

File: build.sbt [hunk 2]
```diff
@@ -90,15 +90,17 @@ lazy val codeGen = projectMatrix
...
+      "com.thesamet.scalapb.common-protos" %% "pgv-proto-scalapb_0.11" % (pgvVersion + "-6") % "protobuf",
+      "com.thesamet.scalapb.common-protos" %% "pgv-proto-scalapb_0.11" % (pgvVersion + "-6")
     ),
     Compile / PB.protoSources += core.base / "src" / "main" / "protobuf",
     Compile / PB.targets := Seq(
       PB.gens.java -> (Compile / sourceManaged).value / "scalapb"
-    )
+    ),
+    // avoids "Not found: type UnusedPrivateParameter" in scalapb/scalapb/validate/Validate.java with Scala 3
+    Compile / compileOrder := CompileOrder.JavaThenScala
```
Comment (bjaglin) [inline build.sbt:101]: Looks like bug in dotty. Without this:
```
[info] compiling 19 Scala sources and 3 Java sources to /home/bjaglin/git/projects/scalapb-validate/code-gen/target/jvm-3/classes ...
[error] -- [E006] Not Found Error: /home/bjaglin/git/projects/scalapb-validate/code-gen/target/jvm-3/src_managed/main/scalapb/scalapb/validate/Validate.java:152:8 
[error] 152 |        UnusedPrivateParameter unused) {
[error]     |        ^^^^^^^^^^^^^^^^^^^^^^
[error]     |        Not found: type UnusedPrivateParameter
[error] -- [E006] Not Found Error: /home/bjaglin/git/projects/scalapb-validate/code-gen/target/jvm-3/src_managed/main/scalapb/scalapb/validate/Validate.java:1209:8 
[error] 1209 |        UnusedPrivateParameter unused) {
[error]      |        ^^^^^^^^^^^^^^^^^^^^^^
[error]      |        Not found: type UnusedPrivateParameter
[error] -- [E006] Not Found Error: /home/bjaglin/git/projects/scalapb-validate/code-gen/target/jvm-3/src_managed/main/scalapb/scalapb/validate/Validate.java:1834:8 
[error] 1834 |        UnusedPrivateParameter unused) {
[error]      |        ^^^^^^^^^^^^^^^^^^^^^^
[error]      |        Not found: type UnusedPrivateParameter
[error] -- [E006] Not Found Error: /home/bjaglin/git/projects/scalapb-validate/code-gen/target/jvm-3/src_managed/main/scalapb/scalapb/validate/Validate.java:2386:8 
[error] 2386 |        UnusedPrivateParameter unused) {
[error]      |        ^^^^^^^^^^^^^^^^^^^^^^
[error]      |        Not found: type UnusedPrivateParameter
```

Comment (bjaglin) [reply build.sbt:101]: It looks like this static inner class is a recurring source of problems https://github.com/scala/bug/issues/9111#issuecomment-501575819

Comment (bjaglin) [reply build.sbt:101]: https://github.com/scalapb/ScalaPB/commit/c4f9ce9198e670310202f3fc744449b85491901b#diff-dd3701b993fcc974a9a6d017159aac00d4b41bcd05ad319815c6001311c669d5R66

**Comment by thesamet** [comment]
Thanks, this got in via 03da6a7

**Pull request closed without merging**